core/src/main/java/hudson/model/View.java

```diff
@@ -463,9 +463,18 @@ private boolean isRelevant(Collection<Label> labels, Computer computer) {
         Collection<TopLevelItem> items = getItems();
         List<Queue.Item> result = new ArrayList<Queue.Item>();
         for (Queue.Item qi : base) {
-            if (items.contains(qi.task)) {
-                result.add(qi);
-            } else
+            // Check if the task of parent tasks are in the list of items.
+            // Pipeline jobs and other jobs which allow parts require us to
+            // check owner tasks as well.
+            Queue.Task currentTask = null;
+            do {
+                currentTask = currentTask == null ? qi.task : currentTask.getOwnerTask();
+                if (items.contains(currentTask)) {
+                    result.add(qi);
+                    break;
+                }
+            } while (currentTask.getOwnerTask() != currentTask);
```

Comment:
I wonder how likely a bug is that results in an infinite loop here. Still a bug elsewhere (`this.getOwnerTask() == this`, but I think we should break out after 10 or so iterations and log a warning.

Reply:
I see no reason for concern. If you override `getOwnerTask` you obviously need to ensure that the chain terminates.

IIUC this loop could be written as

```java
Queue.Task currentTask = ci.task;
while (true) {
    if (items.contains(currentTask)) {
        result.add(qi);
        break;
    }
    Queue.Task next = currentTask.getOwnerTask();
    if (next == currentTask) {
        break;
    } else {
        currentTask = next;
    }
}
```

which may or may not be easier to grasp.

Reply:
As I said in my other comment, the logic is pretty complex, I will try to redesign!

Reply:
> obviously

Bug report coming soon to a `robustness` label near you 😉

Reply:
BTW @daniel-beck how do I log a warning, create static Logger object in View class and then log to warning on that object? Or is there another way?